### ISSUE TITLE
GUI error: error message shown when editing action to new context

### ISSUE THREAD

**Issue by C-Otto** [TracksApp/tracks#1836]
In the project view, when editing an action so that it belongs to a previously non-existing context, an error message "There was an error retrieving from server: error" is shown and the edit popup does not close. However, the new context is created and the action is updated correctly.

This does NOT happen in the starred view.

```
ActionView::Template::Error (undefined method `id' for nil:NilClass):
    2: @not_done = @not_done_todos.select {|t| t.context_id == context.id }
    3: # invalidate the cache every day because of staleness or
    4: # rendering of "due in x days" that change without touching updated at of the todo
    5: cache [context, source_view_key, current_user.date.strftime("%Y%m%d"), @tag_name] do
    6: -%>
    7:   <%=
    8:     render :partial => 'todos/collection',
  app/helpers/application_helper.rb:236:in `eval'
  app/helpers/application_helper.rb:236:in `eval'
  app/helpers/application_helper.rb:236:in `source_view_key'
  app/views/contexts/_context.html.erb:5:in `_app_views_contexts__context_html_erb__2233859530331264606_62088620'
  app/views/todos/_update_successful.js.erb:90:in `_app_views_todos__update_successful_js_erb___2350987662880956348_49463840'
  app/views/todos/update.js.erb:3:in `_app_views_todos_update_js_erb___2064830107471052161_49361520'
```

**Comment by dnrce** [comment]
This is a regression.

**Comment by dnrce** [comment]
Fixed by #1875
